src/main/java/mate/academy/bookstore/service/impl/OrderServiceImpl.java

```diff
@@ -0,0 +1,134 @@
+import mate.academy.bookstore.dto.OrderDto;
+import mate.academy.bookstore.dto.OrderItemDto;
+import mate.academy.bookstore.dto.OrderUpdateRequestDto;
+import mate.academy.bookstore.exception.EntityNotFoundException;
+import mate.academy.bookstore.mapper.OrderItemMapper;
+import mate.academy.bookstore.mapper.OrderMapper;
+import mate.academy.bookstore.model.Book;
+import mate.academy.bookstore.model.CartItem;
+import mate.academy.bookstore.model.Order;
+import mate.academy.bookstore.model.OrderItem;
+import mate.academy.bookstore.model.ShoppingCart;
+import mate.academy.bookstore.model.User;
+import mate.academy.bookstore.repository.OrderItemRepository;
+import mate.academy.bookstore.repository.OrderRepository;
+import mate.academy.bookstore.repository.ShoppingCartRepository;
+import mate.academy.bookstore.service.LocaleService;
+import mate.academy.bookstore.service.OrderService;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class OrderServiceImpl implements OrderService {
+    private final OrderRepository orderRepository;
+    private final OrderItemRepository orderItemRepository;
+    private final OrderMapper orderMapper;
+    private final OrderItemMapper orderItemMapper;
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final LocaleService localeService;
+
+    @Override
+    public OrderDto getOrderById(Long orderId, User user) {
+        Order order = orderRepository.findByIdAndUser_Id(orderId, user.getId())
+                .orElseThrow(() -> new EntityNotFoundException(
+                        localeService.getMessage("exception.notfound.order") + orderId
+                ));
+        return orderMapper.toDto(order);
+    }
+
+    @Override
+    public OrderItemDto getOrderItemById(Long itemId, Long orderId, User user) {
+        OrderItem orderItem = orderItemRepository
+                .findByIdAndOrder_IdAndOrder_User_Id(itemId, orderId, user.getId())
+                .orElseThrow(() -> new EntityNotFoundException(
+                        localeService.getMessage("exception.notfound.orderitem") + itemId
+                ));
+        return orderItemMapper.toDto(orderItem);
+    }
+
+    @Override
+    public List<OrderItemDto> getOrderItemsByOrderId(Long orderId, User user) {
+        List<OrderItem> orderItems = orderItemRepository
+                .findByOrder_IdAndOrder_User_Id(orderId, user.getId());
+        if (orderItems.isEmpty()) {
+            throw new EntityNotFoundException(
+                    localeService.getMessage("exception.notfound.orderitems") + orderId
+            );
+        }
+        return orderItems.stream()
+                .map(orderItemMapper::toDto)
+                .toList();
+    }
+
+    @Override
+    public List<OrderDto> findAll(User user) {
+        return orderRepository.findByUser_Id(user.getId()).stream()
+                .map(orderMapper::toDto)
+                .toList();
+    }
```

Comment:
Why in one case empty collection validation is present but in other no?

Reply:
Because it's ok if user doesn't have any orders :)